app/controllers/view_components_controller.rb

```diff
@@ -37,6 +37,10 @@ def previews
     end
   end
 
+  def system_test_entrypoint
+    render file: "./tmp/#{params[:file]}", status: 200
```

Comment:
```suggestion
    return head :not_authorized if Rails.env.production?
    render file: "./tmp/#{params[:file]}", status: 200
```

Can we guard this in production just in case this is accidentally exposed? It may also be worth trying to sanitize `params[:file]` to prevent accidental file system traversal.

Reply:
Great point! Yes, I think both suggestions are ideal to avoid any unintentional security holes

P.S - I'll try to put some effort into getting this fixed up and ready for review again :)